spring-core/src/main/java/org/springframework/aot/hint/support/RuntimeHintsUtils.java

```diff
@@ -92,4 +97,25 @@ public static void registerAnnotationIfNecessary(RuntimeHints hints, MergedAnnot
 		}
 	}
 
+	/**
+	 * Register that the supplied resource should be made available at runtime.
+	 * <p>If the supplied resource is not a {@link ClassPathResource}, it will
+	 * not be registered.
+	 * @param hints the {@link RuntimeHints} instance to use
+	 * @param resource the resource to register
+	 * @throws IllegalArgumentException if the supplied resource does not
+	 * {@linkplain Resource#exists() exist}
+	 * @see ResourceHints#registerPattern(String)
+	 */
+	public static void registerResource(RuntimeHints hints, Resource resource) {
+		if (resource instanceof ClassPathResource classPathResource) {
+			Assert.isTrue(resource.exists(), () -> "Resource does not exist: " + resource);
+			hints.resources().registerPattern(classPathResource.getPath());
+		}
+	}
+
+	public static void registerResources(RuntimeHints hints, Resource... resources) {
```

Comment:
We didn't have a `registerAnnotations` either. I am not strongly opposed of adding this, but rather than dong this, you could return a `Consumer<ResourcePatternHints.Builder>` and use the existing `registerPattern`.

Reply:
> you could return a Consumer<ResourcePatternHints.Builder> and use the existing registerPattern.

Which code is the "you" in this context?

In any case, I may omit the `registerResources()` method initially. It's a nice "convenience method" to have but not essential.